cpp/src/connection.cpp

```diff
@@ -192,6 +193,15 @@ std::vector<symbol> connection::desired_capabilities() const {
     return get_multiple<std::vector<symbol> >(caps);
 }
 
+std::map<symbol, value> connection::properties() const {
+    std::map<symbol, value> props_ret;
+    value props(pn_connection_remote_properties(pn_object()));
+    if (!props.empty()) {
+        get(props, props_ret);
+    }
+    return props_ret;
+}
+
```

Comment:
I agree this is as efficient as my version - My preference is for multiple returns over (IMO) meaningless local variables. But just my preference - this is fine.

Reply:
Thanks